Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugSystemComponent.cpp

```diff
@@ -440,67 +477,238 @@ namespace Multiplayer
+                    }
 
+                    if (m_hierarchyDebugger)
+                    {
+                        m_hierarchyDebugger->OnImGuiUpdate();
+                    }
+                }
+                ImGui::End();
+            }
+            else
+            {
                 if (m_hierarchyDebugger)
                 {
-                    m_hierarchyDebugger->OnImGuiUpdate();
+                    m_hierarchyDebugger.reset();
+                }
+            }
+
+            if (m_displayNetAuditTrail)
+            {
+                if (ImGui::Begin("Multiplayer Audit Trail", &m_displayNetAuditTrail))
+                {
+                    if (m_auditTrail == nullptr)
+                    {
+                        m_lastFilter = "";
+                        m_auditTrail = AZStd::make_unique<MultiplayerDebugAuditTrail>();
+                        m_committedAuditTrail = m_pendingAuditTrail;
+                    }
+
+                    if (m_auditTrail->CanPumpAuditTrail())
+                    {
+                        m_committedAuditTrail = m_pendingAuditTrail;
+                    }
+
+                    FilterAuditTrail();
+
+                    if (m_auditTrail)
+                    {
+                        if (m_filteredAuditTrail.size() > 0)
+                        {
+                            m_auditTrail->OnImGuiUpdate(m_filteredAuditTrail);
+                        }
+                        else
+                        {
+                            m_auditTrail->OnImGuiUpdate(m_committedAuditTrail);
+                        }
+                    }
+                }
+                ImGui::End();
+            }
+            else
+            {
+                if (m_auditTrail)
+                {
+                    m_auditTrail.reset();
                 }
             }
-            ImGui::End();
         }
-        else
+        else if (m_previousSystemCursorState != AzFramework::SystemCursorState::Unknown)
         {
-            if (m_hierarchyDebugger)
+            AzFramework::InputSystemCursorRequestBus::Event(
+                AzFramework::InputDeviceMouse::Id, &AzFramework::InputSystemCursorRequests::SetSystemCursorState,
+                m_previousSystemCursorState);
+            m_previousSystemCursorState = AzFramework::SystemCursorState::Unknown;
+        }
+    }
+#endif
+
+    void MultiplayerDebugSystemComponent::FilterAuditTrail()
+    {
+        AZStd::string filter = m_auditTrail->GetAuditTrialFilter();
+        if (m_filteredAuditTrail.size() > 0 && filter == m_lastFilter)
+        {
+            return;
+        }
+        m_lastFilter = filter;
+        m_filteredAuditTrail.clear();
+
+        if (filter.size() == 0)
+        {
+            return;
+        }
+
+        for (auto elem = m_committedAuditTrail.begin(); elem != m_committedAuditTrail.end(); ++elem)
+        {
+            const char* nodeTitle = elem->m_category == AuditCategory::Desync
```

Comment:
Nit: consider using params make nested ternarys easier to read / or split up to aid maintainers.